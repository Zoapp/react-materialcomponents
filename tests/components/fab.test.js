import React from "react";
import { shallow } from "enzyme";
import Fab from "libs/components/fab";

describe("components/Fab", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(
      <Fab>
        <div className="child" />
      </Fab>,
    );
  });

  it("can have children", () => {
    expect(wrapper.contains(<div className="child" />)).toEqual(true);
  });

  it("renders an icon if supplied", () => {
    wrapper.setProps({ icon: "favorite" });
    expect(wrapper.find("button .mdc-fab__icon")).toHaveLength(1);
  });

  it("can have a custom label", () => {
    wrapper.setProps({ label: "bar" });
    expect(wrapper.prop("aria-label")).toEqual("bar");
  });
});
