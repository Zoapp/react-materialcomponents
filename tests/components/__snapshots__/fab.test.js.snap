// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Fab can have a custom label 1`] = `
<button
  aria-label="bar"
  className="mdc-fab material-icons rmdc-fab--absolute"
  onClick={null}
/>
`;

exports[`components/Fab can have children 1`] = `
<button
  aria-label={null}
  className="mdc-fab material-icons rmdc-fab--absolute"
  onClick={null}
>
  Children are here !
</button>
`;

exports[`components/Fab renders an icon if supplied 1`] = `
<button
  aria-label={null}
  className="mdc-fab material-icons rmdc-fab--absolute"
  onClick={null}
>
  <span
    className="mdc-fab__icon"
  >
    favorite
  </span>
</button>
`;
